test/integration/node-specific/topology.test.js

```diff
@@ -10,12 +10,20 @@ describe('Topology', function () {
       const states = [];
       topology.on('stateChanged', (_, newState) => states.push(newState));
       topology.connect(err => {
-        expect(err).to.not.exist;
-        topology.close(err => {
+        try {
           expect(err).to.not.exist;
-          expect(topology.isDestroyed()).to.be.true;
-          expect(states).to.eql(['connecting', 'connected', 'closing', 'closed']);
-          done();
+        } catch (error) {
+          done(error);
+        }
+        topology.close({}, err => {
+          try {
+            expect(err).to.not.exist;
+            expect(topology.isDestroyed()).to.be.true;
+            expect(states).to.eql(['connecting', 'connected', 'closing', 'closed']);
+            done();
+          } catch (error) {
+            done(error);
+          }
```

Comment:
Since toplogy.close/connect are internal we're likely to async-ify them in the driver which would cascade to fixing tests that are directly testing these internal methods (instead of via the client). That will cause this test to need re-modification later so possibly not worth changing? 
anyway, happy to update to async/await if we want that or just revert the change that was made here, this is just the classic problem of test times out instead of reporting the real error

Reply:
Agreed, the change here was a stopgap to fix the control flow of this test, but this can definitely be improved.